src/attr_file.c

```diff
@@ -397,7 +397,7 @@ bool git_attr_fnmatch__match(
 
 		/* for attribute checks or root ignore checks, fail match */
 		if (!(match->flags & GIT_ATTR_FNMATCH_IGNORE) ||
-			path->basename == path->path)
+			path->basename == relpath)
```

Comment:
So just to clear up my thinking here:

a) IF we're trying to match a directory
b) AND the path we're looking at is not a directory
c) AND the basename and path are not the same

then we have a path containing at least one directory, e.g. "foo/bar", where we have to do additional checks. If we're inside of "foo" and try to match a directory "bar" but find a file "bar", we should not match, but currently we do not realize because we use the root-based `path->path` ("foo/bar") instead of `relpath` ("bar") to match against the basename.

So I think this change is correct. The comment is becoming a bit stale though, as strictly speaking we do not do this check for the root directory only.